driver-sync/src/main/com/mongodb/client/internal/CollectionInfoRetriever.java

```diff
@@ -33,9 +34,8 @@ class CollectionInfoRetriever {
         this.client = notNull("client", client);
     }
 
-    @Nullable
-    public BsonDocument filter(final String databaseName, final BsonDocument filter, @Nullable final Timeout operationTimeout) {
-        return databaseWithTimeout(client.getDatabase(databaseName), TIMEOUT_ERROR_MESSAGE,
-                operationTimeout).listCollections(BsonDocument.class).filter(filter).first();
+    public ListCollectionsIterable<BsonDocument> filter(final String databaseName, final BsonDocument filter, @Nullable final Timeout operationTimeout) {
```

Comment:
It seems more consistent with the original signature to return `List<BsonDocument>`, as a `ListCollectionsIterable<BsonDocument` both exposes more implementation detail than necessary, and also hasn't actually executed the operation yet.

Reply:
I think async looks right, since you just changed Mono to Flux.